src/Parser/Desugar.ml

```diff
@@ -500,12 +508,22 @@ and tr_expr (e : Raw.expr) =
     make (EMatch(tr_expr e, [make cl1; make cl2]))
   | ESelect(path, e) ->
     make (EDefs([ make (DOpen(false, path)) ], tr_expr e))
-  | EBOp(exp1,op,exp2) -> 
-    let e1 = tr_expr exp1 and e2 = tr_expr exp2 in 
-    make (EApp(
-      { pos  = Position.join e1.pos op.pos; 
-        data = EApp(tr_bop_to_expr op, e1)
-      }, e2))
+  | EBOp(exp1,op,exp2) ->
+    let with_nowhere data = { pos = Position.nowhere; data = data} in
+    begin match op.data with
+    | "&&" ->
+      tr_expr (make (Raw.EIf(exp1, exp2, with_nowhere (Raw.ECtor("False")))))
+    | "||" ->
+      tr_expr (make (Raw.EIf(exp1, with_nowhere (Raw.ECtor("True")), exp2)))
+    | ";" ->
+      tr_expr (make (Raw.EDefs([make (Raw.DLet(false, make Raw.EWildcard, exp1))], exp2)))
```

Comment:
This way of desugaring does not enforce `exp1` to have `Unit` type. Because forgetting function parameter is a common programmer error, it should be fixed.
```
$ cat test3.dbl
let _ = 42; True
$ dbl test3.dbl
$
```

Moreover, this line does not fit 80 columns.